README.md

```diff
@@ -1,51 +1,159 @@
 
-Please install [mbed CLI](https://github.com/ARMmbed/mbed-cli#installing-mbed-cli).
+You can build this project with all supported [Mbed OS build tools](https://os.mbed.com/docs/mbed-os/latest/tools/index.html). However, this example project specifically refers to the command-line interface tool [Arm Mbed CLI](https://github.com/ARMmbed/mbed-cli#installing-mbed-cli).
 
-## Import the example application
+## Application functionality
 
-From the command-line, import the example:
+The `main()` function toggles the state of a digital output connected to an LED on the board.
 
+## Prerequisites
+
+1. Install Mbed CLI.
+1. Determine the toolchain that supports your target.
+
+   Depending on the target, the example project can be built with the GCC_ARM, ARM or IAR toolchain. Run this command to determine which toolchain supports your target:
+
+   ```bash
+   $ mbed compile -S
+   ```
+   
+1. Clone this repository on your system.
+1. Change the current directory to where the project was cloned.
+
+## Building and running
+
+1. Connect a USB cable between the USB port on the target and the host computer.
+1. Run the following command to build the example project and program the microcontroller flash memory:
+
+   ```bash
+   $ mbed compile -m <TARGET> -t <TOOLCHAIN> --flash --sterm
+   ```
+   
+Note: You can use the Mbed CLI command-line option "--sterm" to open a serial terminal after flashing.
+
+Your PC may take a few minutes to compile your code.
+
+The binary is located at `./BUILD/<TARGET>/<TOOLCHAIN>/mbed-os-example-blinky-baremetal.bin`.
+
+Alternatively, you can manually copy the binary to the target, which gets mounted on the host computer through USB.
+
+## Expected output 
+
+The LED on your target turns on and off every 500 milliseconds and the serial terminal shows an output similar to: 
+
+``` 
+--- Terminal on /dev/tty.usbmodem21102 - 9600,8,N,1 ---
+This is the bare metal blinky Mbed OS example.
+``` 
+
+## Configuring the application
+
+### The bare metal mode
+
+The bare metal mode is a configuration of Mbed OS that excludes the RTOS as well as other features. It is specifically targeted for memory constrained devices because it gives developers higher degree of control of their system. See more details [here](https://os.mbed.com/docs/mbed-os/v5.15/reference/mbed-os-bare-metal.html)
+
+To build with the bare metal profile, the application configuration file must contain the following:
 ```
-mbed import mbed-os-example-blinky-baremetal
-cd mbed-os-example-blinky-baremetal
+{
+    "requires": ["bare-metal"]
+}
 ```
 
-### Now compile
+Note: the `mbed_app.json` may be present at the root of your application or specified as the argument of the --app-config parameter. For convenience, a few configuration files are present in the `configs` directory so you can try some of the different configurations shown in this example.
```

Comment:
```suggestion
Note: the application configuration is picked up by the build tool if named `mbed_app.json`. Alternatively, it can be given a more descriptive name and provided to the build tool using an optional argument to the `compile` sub-command. See [here](link to Memory comparison section) for how to provide a different application configuration file to the build tool.
```
Something like that.
Reasons for change request:
1. `mbed_app.json` does not have to be in the root, it gets picked up anywhere if it is called `mbed_app.json` (https://os.mbed.com/docs/mbed-os/v5.15/tools/greentea-testing-applications.html see section `Application configuration`)
2. `--app-config` is mentioned further down, I suggest we remove it from here to avoid repetition.

Reply:
The sentence was directly taken from our documentation https://os.mbed.com/docs/mbed-os/v5.15/reference/configuration.html. I don't want this to become a discussion on configuration. I will just remove the first sentence to remove the repetition on `--app-config`